### PR TITLE
Position ModifiedString and small related fixes

### PR DESCRIPTION
**Pull request by esiragusa**
* Implemented a lightweight Position ModifiedString
* Fixed CountingIterator
* Removed ConstParameter_<>
* Added getter stringSetPositions() to Segment StringSet
* Use Pointer_<> for inner host of Segment StringSet
* Wrapped more parallel STL algorithms

### CODE REVIEW

File: include/seqan/modifier/modifier_position.h [hunk 1]
```diff
@@ -0,0 +1,513 @@
...
+
+//template <typename THost, typename TPositions>
+//struct Suffix<ModifiedString<THost, ModPos<TPositions> > >
+//{
+//    typedef ModifiedString<THost, ModPos<TPositions> >    Type;
+//};
+//
+//template <typename THost, typename TPositions>
+//struct Suffix<ModifiedString<THost, ModPos<TPositions> > const> :
+//    Suffix<ModifiedString<THost, ModPos<TPositions> > > {};
```
Comment (rrahn) [inline include/seqan/modifier/modifier_position.h]: Commented out intensionally, or can it be removed? If not please document why it is commented out.

Comment (esiragusa) [inline include/seqan/modifier/modifier_position.h]: It must be implemented...

File: include/seqan/modifier/modifier_position.h [hunk 2]
```diff
@@ -0,0 +1,513 @@
...
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ==========================================================================
+// Author: Enrico Siragusa <enrico.siragusa@fu-berlin.de>
+// ==========================================================================
+
```
Comment (rrahn) [inline include/seqan/modifier/modifier_position.h]: Could you pleas add a small description of what this class is all about? What it does and for what it is used?

**Comment by rrahn** [comment]
lgtm. Can be merged.

**Pull request merged**